dts/formatting hex values to lowercase

kylebonnici opened the pull request:
This PR automates https://github.com/zephyrproject-rtos/zephyr/pull/100361

commit ccc450ce19a5012f17fcd194b9999ab50db6e0b3 should be ignored as this is part of https://github.com/zephyrproject-rtos/zephyr/pull/100247

CC @erwango

DBS06 reviewed (dismissed):
I can give my approve for the little fuel-gauge part

mathieuchopstm commented on boards/amd/versalnet_apu/versalnet_apu-qemu.dts:
```diff
@@ -1564,7 +1564,7 @@
 			};
 
-			pki_rng@0x20400040000ULL {
+			pki_rng@0x20400040000ull {
```
Isn't this file generated by `qemu` (and never used by Zephyr!) though? I think autogenerated files should not be further formatted.

jfischer-no commented on boards/hardkernel/odroid_go/odroid_go_procpu.dts:
```diff
@@ -155,6 +155,7 @@
 			disk-name = "SD";
 			status = "okay";
 		};
+
```
>Formatting: Further refinement to dts-linter

Use dts: for treewide devicetree changes, https://docs.zephyrproject.org/latest/contribute/guidelines.html#pull-request-guidelines

kylebonnici replied:
I guess this comment was intend for this PR that has now been merged https://github.com/zephyrproject-rtos/zephyr/pull/100247/commits. Noted will be more descriptive in my commit titles

kylebonnici committed:
CI: Update dts-linter dts-linter 0.3.9

Update DTS linter apply lowercase hex values in DTS files
Improve compliance with DTS Coding Style which says that:

4) Hex values in properties, e.g. “reg”, shall use lowercase hex.
   The address part can be padded with leading zeros.

This need dts-linter 0.3.9

Signed-off-by: Kyle Bonnici <kylebonnici@hotmail.com>

kylebonnici committed:
dts: use lowercase hex values in DTS files.

Apply chnages from dts-linter 0.3.9.

Improve compliance with DTS Coding Style which says that:

4) Hex values in properties, e.g. “reg”, shall use lowercase hex.
   The address part can be padded with leading zeros.

Signed-off-by: Kyle Bonnici <kylebonnici@hotmail.com>

MaureenHelm reviewed (dismissed):
> This PR is now blocking 2 other PRs
> 
> * [Formatting/dts indent expression #101116](https://github.com/zephyrproject-rtos/zephyr/pull/101116)
> * [Formatting/no indent for include #101619](https://github.com/zephyrproject-rtos/zephyr/pull/101619)
> 
> is there any action I can take on this PR?

The assignee needs to approve before it can be merged. @de-nordic can you take a look?

You can escalate on the Discord #pr-help channel, but given most people are out this week for the holidays, you might have better luck waiting until next week.

kartben commented:
@jfischer-no please re-visit. I believe your request for change is no longer applicable.

kylebonnici commented on boards/adafruit/feather_esp32s2/adafruit_feather_esp32s2_tft.dts:
Done

kylebonnici commented on boards/amd/versal2_rpu/versal2_rpu-qemu.dts:
```diff
@@ -5797,7 +5797,7 @@ Cannot continue.\nTry installing libgcrypt.";
 		phandle = <0xa2>;
 	};
 
-	ddr_2@0x800000000ULL {
```
I see you have read https://github.com/zephyrproject-rtos/zephyr/pull/100388#discussion_r2584565529 

If we have a preference then I will  update linter to format this way.  BUT IMO `@0x` and `ULL`  in a node name do not have any meaning and as node size is dependent on the `#address-cells`hence `ULL`adds no value and node address are always in hex so `0x` is does not add any value. 

I was planning to do another rule to remove these from files end we end up with `ddr_2@800000000 ` in another PR

kylebonnici replied:
I can also do the above in the same PR if necessary

de-nordic reviewed (approved):
Discussed offline. Acking, to not block the huge PR further, on condition that the `ull` will get removed as proposed by @kylebonnici

pdgendt commented on scripts/dts/python-devicetree/tests/test.dts:
I wonder why changing this file didn't trigger the GH DTS workflow?

```yaml
  pull_request:
    branches:
    - main
    - v*-branch
    paths:
    - 'scripts/dts/**'
    - '.github/workflows/devicetree_checks.yml'
```

kylebonnici replied:
So you are asking why was action not triggered when PR was merged to main?

kylebonnici replied:
I still do not follow after reading https://docs.github.com/en/actions/reference/workflows-and-actions/workflow-syntax#onpull_requestpull_request_targetbranchesbranches-ignore

> Use the branches filter when you want to include branch name patterns or when you want to both include 

only main and v*-branch are included and my branch name was `formatting/dts-lowercase-hex`

> If you define both branches/branches-ignore and [paths/paths-ignore](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore), the workflow will only run when both filters are satisfied.

We do have both branches and paths, we satisfy the path but not the branches hence IMO is correctly did not run.
